pkg/api/types.go

```diff
@@ -303,6 +303,7 @@ type PodSpec struct {
 	// +optional
 	SetHostnameAsFQDN *bool `json:"setHostnameAsFQDN,omitempty"`
 
+	Architecture Architecture `json:"architecture"`
 }
```

Comment:
I can add methods to encode & decode the architecture type as strings if API niceties is important.

Or shall I not include it in the JSON?